draco/synthesis/expand_perturbed.py

```diff
@@ -0,0 +1,598 @@
+
+"""Tasks for expanding sidereal stream data with perturbed beams.
+
+A typical pattern would be to turn a sidereal stream from the :class:`SimulateSidereal` task,
+then generate one or more non-zero perturbation values with the :class:`GeneratePerturbation`
+or :class:`GenerateSinglePerturbation` tasks then apply the perturbations and expand
+nominally redundant products to first order with :class:`ExpandPerturbedProducts`
+or to second order with :class:`ExpandPerturbedProducts2ndOrder`. Optionally, individual
+components of the expanded sidereal stream can be output via :class:`OutputPertStructure`.
+Then, the expanded stream can be passed on to make timestreams as in unperturbed sims.
+
+Tasks
+=====
+
+.. autosummary::
+    :toctree:
+
+    GeneratePerturbation
+    GenerateSinglePerturbation
+    ExpandPerturbedProducts
+    ExpandPerturbedProducts2ndOrder
+    OutputPertStructure
+
+"""
+
+import numpy as np
+
+from cora.util import hputil
+from caput import mpiutil, pipeline, config, mpiarray
+
+from ..core import containers, task
+
+
+
+class GeneratePerturbation(task.SingleTask):
+    """ Generate small, random number perturbatons to input to the
+        ExpandPerturbedProducts task.
+    """
+
+    # Define multiplier value. This is usually set in the .yaml file used to run the pipeline.
+    # The idea is it sets the overall size of the beam perturbations, allowing
+    # us to ensure smaller values than the usual size of a numpy random number.
+    mult = config.Property(proptype = float, default = 0.01)
+    def setup(self,telescope):
+        """Get a reference to the telescope class.
+
+        Parameters
+        -------------
+        tel : : class: `drift.core.TransitTelescope`
+            Telescope object.
+        """
+        self.telescope = telescope
+
```

Comment:
You mention the parameter 'tel' in the docstring, but the variables is called telescope in the code. Unless I am missing something...